src/module/item/affliction/data.ts

```diff
@@ -1,90 +1,383 @@
+                    initial: "unlimited",
+                }),
+                expiry: new fields.StringField({
+                    choices: ["turn-start", "turn-end", "round-end"],
+                    required: true,
+                    nullable: true,
+                    initial: null,
+                }),
+            }),
+            start: new fields.SchemaField({
+                value: new fields.NumberField(),
+                initiative: new fields.NumberField({ required: false, nullable: true, initial: null }),
+            }),
+            context: new EffectContextField(),
+        };
+    }
 
-interface AfflictionSystemSource extends AbstractEffectSystemSource {
-    level: { value: number };
-    save: {
-        type: SaveType;
-        value: number;
-    };
-    stage: number;
-    stages: Record<string, AfflictionStageData>;
-    onset?: AfflictionOnset;
-    duration: DurationData;
-    start: {
-        value: number;
-        initiative: number | null;
-    };
-    /** Origin, target, and roll context of the action that spawned this effect */
-    context: EffectContextData | null;
+    override prepareBaseData(): void {
+        const maxStage = this.stages.length || 1;
+        this.status.stage = Math.clamp(this.status.stage, 1, maxStage);
+    }
 }
 
 interface AfflictionSystemData
-    extends Omit<AfflictionSystemSource, "description" | "fromSpell">,
-        Omit<AbstractEffectSystemData, "level"> {}
-
-interface AfflictionOnset {
-    /** If true, the affliction is currently in its onset phase */
-    active: boolean;
-    value: number;
-    unit: TimeUnit;
-}
+    extends ItemSystemModel<AfflictionPF2e, AfflictionSystemSchema>,
+        Omit<ModelPropsFromSchema<AfflictionSystemSchema>, "description"> {}
 
-interface AfflictionDamage {
-    formula: string;
-    type: DamageType;
-    category?: DamageCategoryUnique | null;
-}
+type AfflictionSystemSchema = Omit<ItemSystemSchema, "traits"> & {
+    level: fields.SchemaField<{
+        value: fields.NumberField<number, number, true, false, true>;
+    }>;
+    traits: fields.SchemaField<{
+        otherTags: fields.ArrayField<SlugField<true, false, false>, string[], string[], true, false, true>;
+        value: fields.ArrayField<
+            fields.StringField<EffectTrait, EffectTrait, true, false, false>,
+            EffectTrait[],
+            EffectTrait[],
+            true,
+            false,
+            true
+        >;
+    }>;
+    save: fields.SchemaField<{
+        type: fields.StringField<SaveType, SaveType, true, false, true>;
+        value: fields.NumberField<number, number, true, false, true>;
+    }>;
+    /** The current status of the affliction, including the current stage and whether it is currently in the onset phase */
+    status: fields.SchemaField<{
+        /** If true, this affliction is in the onset stage */
+        onset: fields.BooleanField<boolean, boolean, false, false, true>;
```

Comment:
Just make anything that has an initial value required

Reply:
Done. The original onset type was optional, but in a data model world this doesn't make sense to be optional anyways.